### ISSUE TITLE
`raw_exec` graphs for `cpu` and `memory` are misleading

### ISSUE THREAD

**Issue by shantanugadgil** [hashicorp/nomad#12455]
### Nomad version
Nomad v1.2.6 (a6c6b475db5073e33885377b4a5c733e1161020c)

### Operating system and Environment details
Amazon Linux 2, Ubuntu 16.04/18.04/20.04

### Issue
Misleading CPU and Memory usage graphs when using `raw_exec` tasks.

### Reproduction steps

Configure a `raw_exec` task with a very low value of `cpu` and `memory` like so:

```hcl
      resources {
        cpu    = 32
        memory = 32
      }
```

When the process starts using much more memory than 32 MB and much more CPU than 32 MHz, the graph in the Nomad Gui looks misleading.

#### Expected Result

Maybe the `raw_exec` graphs could be plotted using the fingerprinted values of the host?

#### Actual Result

Screenshot:
![image](https://user-images.githubusercontent.com/2508915/161732616-8283771c-2b0b-4382-8f39-ab09ee0d8eeb.png)

### Job file (if appropriate)

N/A

### Nomad Server logs (if appropriate)

N/A

### Nomad Client logs (if appropriate)

N/A

**Comment by shoenig** [comment]
Thanks for the report, @shantanugadgil ! Indeed this seems fairly trivial to reproduce

<details><summary>example jobfile</summary>

```hcl
job "py" {
  datacenters = ["dc1"]
  type        = "service"

  group "py" {
    task "py" {
      driver = "raw_exec"
      user   = "nobody"
      config {
        command = "/usr/bin/python3"
        args    = ["-m", "http.server", "--directory", "/tmp"]
      }
      resources {
        memory = 12
        cpu    = 12
      }
    }
  }
}
```

</details>

Just run that and browse to the task in the UI.

**Comment by Lord-Y** [comment]
I'm having kind of the same problem with nomad 1.2.6 with docker image.

![image](https://user-images.githubusercontent.com/8832734/164403039-71b01cd9-a796-4d15-8597-ce2d6dc17f11.png)

The memory usage is always having 0 bytes while docker stats tell totally the opposite:

```
CONTAINER ID   NAME                                          CPU %     MEM USAGE / LIMIT   MEM %     NET I/O          BLOCK I/O         PIDS
9474873dc105   server-6eb853a9-8ad0-be54-edca-1b4a091b1a16   13.58%    252.7MiB / 256MiB   98.69%    72.5MB / 203MB   35.4MB / 50.3MB   48
```
In our job spec we reserve 256MB with no limit.
OS:
```
Distributor ID: Debian
Description:    Debian GNU/Linux 11 (bullseye)
Release:        11
Codename:       bullseye
```

We have no particular error in agent log in debug.

**Comment by Lord-Y** [comment]
Okay so I can see that my memory graph issue is related to cgroup v2 and will be fixed in nomad 1.3.0. More informations are here https://github.com/hashicorp/nomad/issues/10251. I'll wait for nomad 1.3.0

**Comment by shantanugadgil** [comment]
closing the issue due to age.
